lib/valueflow.cpp

```diff
@@ -1692,52 +1725,127 @@ static bool valueFlowForward(Token * const               startToken,
+static bool isStdMoveOrStdForwarded(Token * tok, ValueFlow::Value::MoveKind * moveKind, Token ** varTok = nullptr)
+{
+    bool isMovedOrForwarded = false;
+    ValueFlow::Value::MoveKind kind = ValueFlow::Value::MovedVariable;
+    Token * variableToken = nullptr;
+    if (Token::Match(tok, "std :: move ( %var% )")) {
+        variableToken = tok->tokAt(4);
+        isMovedOrForwarded = true;
+        kind = ValueFlow::Value::MovedVariable;
+    } else if (Token::simpleMatch(tok, "std :: forward <")) {
+        Token * leftAngle = tok->tokAt(3);
+        Token * rightAngle = leftAngle->link();
+        if (Token::Match(rightAngle, "> ( %var% )")) {
+            variableToken = rightAngle->tokAt(2);
+            isMovedOrForwarded = true;
+            kind = ValueFlow::Value::ForwardedVariable;
+        }
+    }
+    if (isMovedOrForwarded && variableToken->strAt(2) == ".")
+        isMovedOrForwarded = false;  // Only partially moved
+    if (moveKind != nullptr)
+        *moveKind = kind;
+    if (varTok != nullptr)
+        *varTok = variableToken;
+    return isMovedOrForwarded;
+}
+
+static const Token * nextAfterAstRightmostLeaf(Token const * tok)
 {
+    const Token * rightmostLeaf = tok;
+    if (!rightmostLeaf || !rightmostLeaf->astOperand1())
+        return nullptr;
+    do {
+        if (rightmostLeaf->astOperand2())
+            rightmostLeaf = rightmostLeaf->astOperand2();
+        else
+            rightmostLeaf = rightmostLeaf->astOperand1();
+    } while (rightmostLeaf->astOperand1());
+    return rightmostLeaf->next();
+}
+
+static void valueFlowAfterAssign(TokenList *tokenlist, SymbolDatabase* symboldatabase, ErrorLogger *errorLogger, bool isCpp, const Settings *settings)
+{
+    bool isAtLeastCpp11 = isCpp && settings->standards.cpp >= Standards::CPP11;
     const std::size_t functions = symboldatabase->functionScopes.size();
     for (std::size_t i = 0; i < functions; ++i) {
         const Scope * scope = symboldatabase->functionScopes[i];
-        for (Token* tok = const_cast<Token*>(scope->classStart); tok != scope->classEnd; tok = tok->next()) {
-            // Assignment
-            if ((tok->str() != "=") || (tok->astParent()))
-                continue;
-
-            // Lhs should be a variable
-            if (!tok->astOperand1() || !tok->astOperand1()->varId())
-                continue;
-            const unsigned int varid = tok->astOperand1()->varId();
-            const Variable *var = tok->astOperand1()->variable();
-            if (!var || (!var->isLocal() && !var->isArgument()))
-                continue;
+        const Token * start = scope->classStart;
+        if (scope->function) {
+            const Token * memberInitializationTok = scope->function->constructorMemberInitialization();
+            if (memberInitializationTok)
+                start = memberInitializationTok;
+        }
+
+        std::vector<FlowBarrierType> flowBarriers;
```

Comment:
somehow when I look at it .. it does not look natural that std::move() is handled in valueFlowAfterAssign().

Maybe we could add a valueFlowAfterMove() such as:

    void valueFlowAfterMove() {
        if (!isAtLeastCpp11)
            return;
        for (Token* tok = start; tok != scope->classEnd; tok = tok->next()) {
            if (!isStdMoveOrStdForwarded(tok, &moveKind, &varTok))
                continue;
            // x is not MOVED after assignment if code is:  x = ... std::move(x) .. ;
            const Token *parent = tok->astParent();
            while (parent && parent->str() != "=")
                 parent = parent->astParent();
            if (parent && parent->astOperand2()->str() == varTok->str())
                 continue;
             valueFlowForward...
        }
    }

I am hoping this will mean that we can keep valueFlowAfterAssign as it is.